File: order-service/OrderService/order-service.Infrastructure/Messaging/DlqMessage.cs
namespace order_service.Infrastructure.Messaging;


// Encapsula uma mensagem que chegou na Dead Letter Queue
public sealed record DlqMessage(
    string MessageId,
    string EventType,
    string SourceQueue,
    string RoutingKey,
    string DeathReason, // Motivo pelo qual a mensagem foi rejeitada
    long DeathCount, // Quantas vezes essa mensagem foi rejeitada / expirou
    DateTimeOffset FirstDeathAt,
    ReadOnlyMemory<byte> Body // Corpo original da mensagem em bytes
); 

File: fraud-service/app/messaging/models/dlq_message.py
from dataclasses import dataclass
from datetime import datetime 

@dataclass(slots = True)
class DlqMessage:
    message_id: str
    event_type: str
    source_queue: str
    routing_key: str
    death_reason: str
    death_count: int
    first_death: datetime
    body: str